### ISSUE TITLE
Using dill to pickle a momba object

### ISSUE THREAD

**Issue by landonjefftaylor** [koehlma/momba#6]
We've been trying to pickle momba objects using dill, and we're getting the following error:

ValueError: ('/', <function <lambda> at 0x000002C6602D1F30>) is not a valid ArithmeticBinaryOperator

We're just wondering if you have successfully pickled momba objects, and if you have any pointers.

**Comment by koehlma** [comment]
I did just try to pickle a Momba object – there is some problem with the way the enums for the operators are defined. I suspect this to be caused by the use of `lambda` functions. However, the error message is different than the one you got. Could you provide a minimal example for reproducing the error such that I can further investigate it?

**Comment by landonjefftaylor** [comment]
Apologies for the delayed reply; I was trying to track down where the error was coming from.

When I run the following script:

```python
from momba.model.expressions import ArithmeticBinary
from momba.model.operators import ArithmeticBinaryOperator
import dill as pickle

class VariableState:
    def __init__(self):
        self.probability = ArithmeticBinary(ArithmeticBinaryOperator.REAL_DIV, 1, 6)

var_state = VariableState()

with open("bug.txt", "wb") as writefile:
    writefile.write(pickle.dumps(var_state))

with open("bug.txt", "rb") as readfile:
    new_vars = pickle.load(readfile)

print(str(new_vars))
```

I get the following message back:

```
Traceback (most recent call last):
  File "---\bug.py", line 15, in <module>
    new_vars = pickle.load(readfile)
  File "---\Python\Python310\site-packages\dill\_dill.py", line 313, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "---\Python\Python310\site-packages\dill\_dill.py", line 525, in load
    obj = StockUnpickler.load(self)
  File "---\Python310\lib\enum.py", line 385, in __call__
    return cls.__new__(cls, value)
  File "---\Python310\lib\enum.py", line 710, in __new__
    raise ve_exc
ValueError: ('/', <function <lambda> at 0x000002A69E2CC3A0>) is not a valid ArithmeticBinaryOperator
```

The error comes when it tries to read the object back in (rather than writing it to the file), and it looks like it just can't process the ArithmeticBinaryOperator REAL_DIV. When I just dump the file as in the code below, I get no error.

```python
from momba.model.expressions import ArithmeticBinary
from momba.model.operators import ArithmeticBinaryOperator
import dill as pickle

class VariableState:
    def __init__(self):
        self.probability = ArithmeticBinary(ArithmeticBinaryOperator.REAL_DIV, 1, 6)

var_state = VariableState()

with open("bug.txt", "wb") as writefile:
    writefile.write(pickle.dumps(var_state))
```

**Comment by koehlma** [comment]
Thanks for the example.

Pickling should now work with the most recent Git version of Momba. Could you try to verify that everything works?

I will prepare a release in the next days.

**Comment by landonjefftaylor** [comment]
It's working beautifully on my machine. Thank you!

**Comment by koehlma** [comment]
The few days took longer than anticipated but the fix is now included in version 0.6.2. :)
